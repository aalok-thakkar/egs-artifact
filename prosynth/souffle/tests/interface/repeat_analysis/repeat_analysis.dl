.type Node
.decl edge (node1:Node, node2:Node)
.input edge

.decl source (node:Node)
.input source
.decl sink (node:Node)
.input sink

.decl path_from_source (node1:Node, node2:Node)

path_from_source(X,Y) :-
     source(X),
     edge(X,Y).
path_from_source(X,Z) :-
     path_from_source(X,Y),
     edge(Y,Z).

.decl source2sink(source:Node, sink:Node)
.output source2sink

source2sink(Source,Sink):-
     path_from_source(Source,Sink),
     sink(Sink).
