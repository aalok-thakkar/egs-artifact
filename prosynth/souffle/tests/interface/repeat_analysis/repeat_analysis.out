source2sink - run 1
source2sink - run 2
B-F
C-F
source2sink - run 3
D-F
